StorageSnapshot.Core/Services/LocalStorageDeviceService.cs

```diff
@@ -44,7 +44,20 @@ public async Task<IEnumerable<LocalStorageDevice>> GetLocalStorageDevicesAsync()
 
     private static IEnumerable<LocalStorageDevice> GetDriveInfo()
     {
-        return DriveInfo.GetDrives().Select(x => new LocalStorageDevice(x));
+        var drives = DriveInfo.GetDrives().Select(x => new LocalStorageDevice(x));
+
+        return drives.ToList().Where(DriveIsPermissible);
+    }
+
+    private static bool DriveIsPermissible(LocalStorageDevice drive)
+    {
+        try
+        {
+            _ = drive.VolumeLabel;
```

Comment:
Does RootDirectory.Exists also work? That should be false if access isn't available.

Reply:
https://stackoverflow.com/a/11710169

"If you do not have at a minimum read-only permission to the directory, the [Exists](https://learn.microsoft.com/en-us/dotnet/api/system.io.directory.exists?view=net-8.0) method will return false." - https://learn.microsoft.com/en-us/dotnet/api/system.io.directory.exists?view=net-8.0&redirectedfrom=MSDN#System_IO_Directory_Exists_System_String_

This should work.